### ISSUE TITLE
Pkg.gc on mac fails to ignore .DS_Store

### ISSUE THREAD

**Issue by tlienart** [JuliaLang/Pkg.jl#790]
```
using Pkg
Pkg.gc()
```

on 0.7 and 1.0.1 give me

```
    Active manifests at:
        `/Users/tlienart/.julia/environments/v0.7/Manifest.toml`
        `/Users/tlienart/.julia/environments/v1.0/Manifest.toml`
ERROR: SystemError: unable to read directory /Users/tlienart/.julia/packages/.DS_Store: Not a directory
Stacktrace:
 [1] readdir(::String) at ./file.jl:620
 [2] macro expansion at ./logging.jl:321 [inlined]
 [3] #gc#43(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::Function, ::Pkg.Types.Context) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:326
 [4] gc at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/Pkg/src/API.jl:290 [inlined] (repeats 2 times)
 [5] top-level scope at none:0
```

The `.DS_Store` is a metadata file for folders which can be safely ignored I would think unless I've missed something?

**Comment by fredrikekre** [comment]
#601 which should have been fixed by #634, but maybe the commit is not in Julia 1.0.1?

**Comment by fredrikekre** [comment]
No worries. I can reproduce on Julia 1.0.1 so seems like this has not been fixed.

**Comment by KristofferC** [comment]
Aargh.

**Comment by fredrikekre** [comment]
The fix is on the `release-1.0` branch now, so should be fixed in Julia v1.0.2.
